### PR TITLE
casadm robustness and security fixes

### PR DESCRIPTION
**Pull request by arutk**

### CODE REVIEW

File: casadm/cas_lib.c [hunk 1]
```diff
@@ -159,10 +159,24 @@ cas_printf_t cas_printf = std_printf;
...
+	}
+
+	if (stat(dev_path, &status) == -1) {
+		printf("Failed to query device status.\n");
 		return FAILURE;
 	}
+
+	if (!S_ISBLK(status.st_mode)) {
+		printf("Path does not describe a block device\n");
+		return failure;
```
Comment (Ostrokrzew) [inline casadm/cas_lib.c:177]: FAILURE

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

File: casadm/cas_lib.c (hunk 1 above)
Comment (arutk) [reply casadm/cas_lib.c:177]: thanks

File: casadm/cas_lib.c [hunk 2]
```diff
@@ -2753,8 +2768,11 @@ int list_caches(unsigned int list_format, bool by_id_path)
 		float core_flush_prog;
 
 		if (!by_id_path) {
-			get_dev_path(curr_cache->device, curr_cache->device,
-				     sizeof(curr_cache->device));
+			if (get_dev_path(curr_cache->device, curr_cache->device,
+				     sizeof(curr_cache->device))) {
+				cas_printf(LOG_WARNING, "WARNING: Cannot resolve path "
+					   "to cache. By-id path will be shown for that cache.\n");
+			}
```
Comment (robertbaldyga) [inline casadm/cas_lib.c:2775]: Mixed tabs and spaces.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

File: casadm/cas_lib.c (hunk 2 above)
Comment (arutk) [reply casadm/cas_lib.c]: fixed

**Review by robertbaldyga — APPROVED** [review]

**Review by mmichal10 — APPROVED** [review]

**Pull request merged**